lib/fakturoid/oauth/credentials.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Fakturoid
+  class Oauth
+    class Credentials
+      EXPIRY_BUFFER_IN_SECONDS = 10
+
+      attr_accessor :access_token, :refresh_token, :token_type
+      attr_reader :expires_at
+
+      def initialize(values = {})
+        update(values)
+      end
+
+      def update(values)
+        values = values.transform_keys(&:to_sym)
+
+        self.access_token  = values[:access_token]
+        self.refresh_token = values[:refresh_token] unless Utils.empty?(values[:refresh_token])
+        self.expires_at    = values[:expires_at] || values[:expires_in]
+        self.token_type  ||= values[:token_type]
+      end
+
+      def expires_at=(value)
+        @expires_at =
+          if value.is_a?(String)
+            DateTime.parse(value)
+          elsif value.is_a?(Integer)
+            prepare_expiry_time(value)
+          else
+            value
+          end
+      end
+
+      def expires_in=(value)
+        self.expires_at = value
+      end
+
+      def access_token_near_expiration?
```

Comment:
Actually this method returns `true` if token is already expired, right? 

To refresh access token before it expires, we should e.g. add 2 seconds to current time in comparison. It would cause that we refresh access token that could expire before request is performed on the API server.

Reply:
We already add 10 seconds, it is done in `Credentials` when parsing the string. Now that I think about it, it may not be the smartest thing in the world. It would make more sense to alter this check than to modify the date. Agree?

Reply:
Yep, to me it makes sense to alter this check.